beamer/util.py

```diff
@@ -147,16 +147,22 @@ def make_web3(
 
 
 class TokenMatchChecker:
-    def __init__(self, tokens: List[List[List[str]]]) -> None:
+    def __init__(
+        self, tokens: List[List[List[str]]], source_chain_id: ChainId, target_chain_id: ChainId
+    ) -> None:
         # A mapping of tokens to equivalence classes. Each frozenset contains
         # tokens that are considered mutually equivalent.
         self._tokens: dict[_Token, frozenset[_Token]] = {}
 
+        running_chain_ids = [source_chain_id, target_chain_id]
+
         for token_mapping in tokens:
             equiv_class = frozenset(
                 (ChainId(int(token[0])), to_checksum_address(token[1])) for token in token_mapping
             )
-            l2_chain_ids = frozenset(chain_id for chain_id, _ in equiv_class)
+            l2_chain_ids = frozenset(
```

Comment:
I think the check for l2_chain_ids is not enough, but if we pull the filtering out as suggested above the TokenMatchChecker should just work out of the box and wouldnt require any changes